src/Functions/array/arrayAggregation.cpp

```diff
@@ -185,6 +193,38 @@ struct ArrayAggregateImpl
                         res[i] = x;
                     }
                 }
+                else if constexpr (aggregate_operation == AggregateOperation::product)
+                {
+                    size_t array_size = offsets[i] - pos;
+                    size_t num = array_size;
+                    AggregationType product = x;
+
+                    if constexpr (IsDecimalNumber<Element>)
+                    {
+                        using T = decltype(x.value);
+                        T x_val = x.value;
+
+                        for (i = 1; i < array_size; ++i)
```

Comment:
Is this right behaviour that this i variable is same that is used in outer loop ? PVS studio detected this as error.

Reply:
Yes, I think it is right. Because the first value is saved as `x_val` outside this loop. for example , `array_size = 2`, then I should only multiply ` x_val `and `  product_val` once.